From b24a90bb359e42e1ee2b1906516d0254e323d115 Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Tue, 15 May 2018 16:07:30 +0800
Subject: [PATCH] arm: dts: rk3288: add needed nodes for trust in linux

This patch adds needed nodes for trust in rk3288-linux:

1.Add psci v1.0 support to call into secure world
  through psci APIs.

2.Use non-secure dmac instead of secure one.

3.Use new efuse compatible to match secure interface
  when kernel is in no-secure mode.

Change-Id: Id7fda3144388661979149cfa00822106b2872712
Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-linux.dtsi | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-linux.dtsi b/arch/arm/boot/dts/rk3288-linux.dtsi
index 535813357b8b..c6db1d34a5af 100644
--- a/arch/arm/boot/dts/rk3288-linux.dtsi
+++ b/arch/arm/boot/dts/rk3288-linux.dtsi
@@ -9,4 +9,33 @@
 		bootargs = "earlyprintk console=tty1 console=ttyS2,115200n8 rw root=PARTUUID=614e0000-0000 rootfstype=ext4 rootwait";
 	};
 
+	psci {
+		compatible = "arm,psci-1.0";
+		method = "smc";
+	};
+};
+
+&cpu0 {
+	enable-method = "psci";
+};
+
+&cpu1 {
+	enable-method = "psci";
+};
+
+&cpu2 {
+	enable-method = "psci";
+};
+
+&cpu3 {
+	enable-method = "psci";
+};
+
+&dmac_bus_s {
+	/* change to non-secure dmac */
+	reg = <0x0 0xff600000 0x0 0x4000>;
+};
+
+&efuse {
+	compatible = "rockchip,rk3288-secure-efuse";
 };
-- 
2.35.3

